Userland/Libraries/LibCore/MemoryStream.cpp

```diff
@@ -125,4 +126,120 @@ size_t FixedMemoryStream::remaining() const
     return m_bytes.size() - m_offset;
 }
 
+ErrorOr<Bytes> AllocatingMemoryStream::read(Bytes bytes)
+{
+    size_t read_bytes = 0;
+
+    while (read_bytes < bytes.size()) {
+        VERIFY(m_write_offset >= m_read_offset);
```

Comment:
I don't see how this can keep holding after one iteration?
`m_write_offset` is not updated? But the test is not failing so what am I missing?

Reply:
`next_read_range` will (in theory) only ever return a range that ends at the current write offset or earlier. If we already are at the current write offset, it will return a range of size zero.

Therefore, `m_read_offset + range.size()` should never get bigger than `m_write_offset`. To make sure that we never violate that assumption, we still check that this is the case after every iteration.